src/crypto/mod.rs

```diff
@@ -3,32 +3,31 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment:
This file is probably where the review is needed, I've left comments to guide the review a little bit.